### ISSUE TITLE
aws_security_group_ids data source

### ISSUE THREAD

**Issue by guigo2k** [terraform-providers/terraform-provider-aws#2740]
Hi there,

We would love to have the ability of querying all the security groups in a VPC. The idea is to have a `aws_security_group_ids` data source, equivalent to `aws_subnet_ids`.

I have multiple Kubernetes clusters running on the same VPC, and need to Terraform a VPC peering which will open all `nodes` security groups. Using the `aws_security_group` data source, I can query the security groups, however, it fails if there is more than one match. 

Is there any reason for [constraining the results](https://github.com/terraform-providers/terraform-provider-aws/blob/63b287b4d32163a4909c8b7f866c2922a8651d43/aws/data_source_aws_security_group.go#L83-L85)?

```
data "aws_security_group" "nodes" {
  vpc_id = "${var.vpc_id}"
  tags {
    Name = "*nodes*"
  }
}
```
Produces the following error:
```
Error: Error refreshing state: 1 error(s) occurred:

* data.aws_security_group.nodes: 1 error(s) occurred:

* data.aws_security_group.nodes: data.aws_security_group.nodes: multiple Security Groups matched; use additional constraints to reduce matches to a single Security Group

```

Terraform v0.11.1

**Comment by trung** [comment]
Until the `aws_security_group_ids` comes to life ... I usually apply an workaround (also for other data sources having similar requirement).

Assumption is:
* You have a consistent naming for your tags, like: `node-0`, `node-1` ... 
* You know number of nodes

```hcl
data "aws_security_group" "nodes" {
  count = "${number_of_nodes}"
  vpc_id = "${var.vpc_id}"
  tags {
    Name = "node-${count.index}
  }
}
```

Then you refer to the list of security group ids by `data.aws_security_group.nodes.*.id`

**Comment by guigo2k** [comment]
Thanks for the workaround @trung, will keep this in mind :)

Unfortunately, I don't have consistent naming for my tags. I use Kops for deploying k8s on AWS and it tags all node instances as `nodes.<clustername>` and the same for security groups :(

**Comment by ian-d** [comment]
I took a shot at this in PR #2947, but opted to make a new resource (`aws_security_groups`) instead of changing the behavior of the existing `aws_security_group` data source. It'll also let you filter on the other security group properties exposed by the API [0].

I think this'll fulfill your use case?

[0] https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-security-groups.html

**Comment by bflad** [comment]
The new `aws_security_groups` data source has been merged into master and will release with version 1.25.0 of the AWS provider later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
